### PR TITLE
Update resiliency page

### PR DESCRIPTION
**Pull request by mxm**
This is based on the latest state of https://github.com/elastic/elasticsearch/blob/master/docs/resiliency/index.asciidoc

### CODE REVIEW

File: docs/architecture/resilience.txt [hunk 1]
```diff
@@ -114,6 +114,15 @@ Here are some considerations:
    rare consistency or data-loss issues you might encounter when CrateDB
    experiences network or hardware failures (see next section).
 
+
+Nightly Jepsen tests for dirty reads, lost updates, version divergence
+----------------------------------------------------------------------
+
+We run nightly tests of the Jepsen suite which test dirty reads, lost updates, and version divergence.
+The source code of these tests can be inspected in the official `Jepsen repository <https://github.com/jepsen-io/jepsen>`_.
+The tests are passing.
```
Comment (seut) [inline docs/architecture/resilience.txt]: that is not true, they are currently expected to fail which they do

File: docs/architecture/resilience.txt [hunk 2]
```diff
@@ -290,6 +299,54 @@ matches the lost update. This will break `Optimistic Concurrency Control
...
+    <td>Severity</td>
+    <td>Modest</td>
+   </tr>
+   <tr>
+    <td>Likelihood</td>
+    <td>Rare</td>
+   </tr>
+   <tr>
+    <td>Cause</td>
+    <td>Primary fails and writes only to a subset of its replicas</td>
```
Comment (seut) [inline docs/architecture/resilience.txt]: would rephrase to: The primary shards fails while in-flight writes are not yet written to all replica shards.

File: docs/architecture/resilience.txt (hunk 1 above)
Comment (mxm) [reply docs/architecture/resilience.txt]: I know we don't support all Jepsen tests but I thought the subset described here passes. Also the latest run passes: https://jenkins.crate.io/job/CrateDB/job/crate_jepsen_test/229/ The jepsen.log files in the artifact doesn't report any failure.

File: docs/architecture/resilience.txt (hunk 2 above)
Comment (mxm) [reply docs/architecture/resilience.txt]: +1

File: docs/architecture/resilience.txt (hunk 1 above)
Comment (seut) [reply docs/architecture/resilience.txt]: This job will pass even if the jepsen tests are failing (see https://jenkins.crate.io/job/CrateDB/job/crate_jepsen_test/200). Its not that easy to let them fail in a deterministic way. The main intention of this job is to ensure that our jepsen suite and tests are running without any issues against the current crate version despite of the test results.

**Comment by mxm** [comment]
@seut Does this look good now?

File: docs/architecture/resilience.txt [hunk 3]
```diff
@@ -114,6 +114,14 @@ Here are some considerations:
    rare consistency or data-loss issues you might encounter when CrateDB
    experiences network or hardware failures (see next section).
 
+
+Nightly Jepsen tests for dirty reads, lost updates, version divergence
```
Comment (seut) [inline docs/architecture/resilience.txt]: I would be so specific in what the jepsen suite is testing. this could change anytime and then we'd have update this document always..

File: docs/architecture/resilience.txt [hunk 4]
```diff
@@ -437,6 +493,59 @@ the node with the original primary shard rejoins the cluster, CrateDB has no
 way of distinguishing correct and incorrect data, so that data replaced with
 data from the new primary shard.
 
+Make table creation resilient to closing and full cluster crashes
```
Comment (mfussenegger) [inline docs/architecture/resilience.txt]: Not sure if it makes sense to document issues that have been resolved quite a while ago?

File: docs/architecture/resilience.txt (hunk 3 above)
Comment (mxm) [reply docs/architecture/resilience.txt]: I think we should be specific and it's ok to update this. Otherwise we can drop the entire paragraph.

File: docs/architecture/resilience.txt (hunk 4 above)
Comment (mxm) [reply docs/architecture/resilience.txt]: I wanted to make this complete. Let's drop all resolved issues from the document then.

Comment (seut) [reply docs/architecture/resilience.txt]: I wouldn't drop existing old one as people may know them and want to see the update.

File: docs/architecture/resilience.txt (hunk 3 above)
Comment (seut) [reply docs/architecture/resilience.txt]: Then lets remove it, I don't agree on being so specific here ;)

File: docs/architecture/resilience.txt (hunk 4 above)
Comment (mxm) [reply docs/architecture/resilience.txt]: I specifically made sure to compare all issues with the ES document. I don't feel good with merging this in an incomplete state. I'm ok with removing the "fixed issues" section but not presenting it incomplete.

File: docs/architecture/resilience.txt (hunk 3 above)
Comment (matriv) [reply docs/architecture/resilience.txt]: I like that we mention the Jepsen tests which somehow shows to our users that we care about the resiliency.
On the other hand I also don't like the specific mention of the tests. Personally I would keep it more abstract, but also I wouldn't like to remove the whole paragraph :-(

File: docs/architecture/resilience.txt (hunk 4 above)
Comment (matriv) [reply docs/architecture/resilience.txt]: I don't have any strong opinion on that. If this issue was encountered in our older CrateDB versions, I don't mind to mention it here as fixed.

File: docs/architecture/resilience.txt (hunk 3 above)
Comment (mxm) [reply docs/architecture/resilience.txt]: Removed the test details.

**Comment by mxm** [comment]
@seut Made adjustments in line with your comments. Could you please have another look?
